elements/lisk-chain/src/state_store/account_store.ts

```diff
@@ -156,51 +132,18 @@ export class AccountStore {
 			);
 		}
 
-		const updatedKeys = Object.entries(updatedElement).reduce<string[]>(
-			(existingUpdatedKeys, [key, value]) => {
-				const account = this._data[elementIndex];
-				// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access,@typescript-eslint/no-explicit-any
-				if (!isEqual(value, (account as any)[key])) {
-					existingUpdatedKeys.push(key);
-				}
-
-				return existingUpdatedKeys;
-			},
-			[],
-		);
-
 		this._data[elementIndex] = updatedElement;
-		// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-		this._updatedKeys[elementIndex] = this._updatedKeys[elementIndex]
-			? [...new Set([...this._updatedKeys[elementIndex], ...updatedKeys])]
-			: updatedKeys;
+		this._updatedKeys.add(primaryValue);
 	}
 
-	public async finalize(tx: StorageTransaction): Promise<void> {
-		const affectedAccounts = Object.entries(this._updatedKeys).map(
-			([index, updatedKeys]) => ({
-				updatedItem: this._data[parseInt(index, 10)].toJSON(),
-				updatedKeys,
-			}),
-		);
-
-		const updateToAccounts = affectedAccounts.map(
-			async ({ updatedItem, updatedKeys }) => {
-				const filter = { [this._primaryKey]: updatedItem[this._primaryKey] };
-				const updatedData = updatedKeys.reduce<Partial<AccountJSON>>(
-					(data, key) => {
-						// eslint-disable-next-line
-						(data as any)[key] = (updatedItem as any)[key];
-
-						return data;
-					},
-					{},
+	public finalize(batch: BatchChain): void {
+		for (const account of this._data) {
+			if (this._updatedKeys.has(account.address)) {
+				batch.put(
```

Comment:
is `batch.put` sync?

Reply:
`batch.put` is `sync` and only `batch.write` is async, and that will save all the updated things atomically